### PR TITLE
Cloud function migration issue when update to Google v1 api

### LINKED ISSUE #202: Failed to Update the Existing Cloud Function through v2.4.2
After serverless-google-cloudfunctions upgrade to support v1 api, when use serverless to deploy existing cloud function via v1 api, 
 there will be a upgrade error :

> Resource types cannot be changed, previous (cloudfunctions.v1beta2.function) -> updated (gcp-types/cloudfunctions-v1:projects.locations.functions)

More detail can be found from here
https://github.com/serverless/serverless-google-cloudfunctions/pull/165

### PR DESCRIPTION
**Pull request by bao1018**
This PR is going to fix below issue:
https://github.com/serverless/serverless-google-cloudfunctions/issues/202

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,6598 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (medikoo) [inline yarn.lock]: I believe this file should not be part of a repo

Comment (bao1018) [reply yarn.lock]: @medikoo thanks for the suggestion, I have deleted yarn.lock

**Comment by bao1018** [comment]
@medikoo I am confused from here, why do we need this process ? I think we just release a new version base upon #165 
1. Revert #165
2. Publish v2.4.3
3. Take #165 again
4. Publish as v3.0.0

**Comment by medikoo** [comment]
> @medikoo I am confused from here, I think we just release a new version base upon #165, why do we need this process ?

I believe that #165 was a breaking change, that we've published with patch update, when it should have been published with a major bump.

Or wasn't that a breaking change in your opinion?

Note this package respects [semver](https://semver.org/)

**Comment by bao1018** [comment]
@medikoo we just found some issues in this PR after some regression test, I will close this PR first

**Pull request closed without merging**